enif_select compilation error by C++ compiler

**Describe the bug**

I am getting the following error when compiling my code that calls `enif_select_read` by `g++`:

```bash
.../erlang/26.1.1/erts-14.1/include/erl_nif_api_funcs.h:652:47: error: invalid conversion from ‘int’ to ‘ErlNifSelectFlags’ [-fpermissive]
  652 |     enif_select_x(ENV, E, ERL_NIF_SELECT_READ | ERL_NIF_SELECT_CUSTOM_MSG, \
      |                           ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                               |
      |                                               int
```

Should all OR'd flags in the calls below be typecasted to `(ErlNifSelectFlags)`?
```c
#  define enif_select_read(ENV, E, OBJ, PID, MSG, MSG_ENV) \
    enif_select_x(ENV, E, ERL_NIF_SELECT_READ | ERL_NIF_SELECT_CUSTOM_MSG, \
                  OBJ, PID, MSG, MSG_ENV)
#  define enif_select_write(ENV, E, OBJ, PID, MSG, MSG_ENV) \
    enif_select_x(ENV, E, ERL_NIF_SELECT_WRITE | ERL_NIF_SELECT_CUSTOM_MSG, \
                  OBJ, PID, MSG, MSG_ENV)
#  define enif_select_error(ENV, E, OBJ, PID, MSG, MSG_ENV) \
    enif_select_x(ENV, E, ERL_NIF_SELECT_ERROR | ERL_NIF_SELECT_CUSTOM_MSG, \
                  OBJ, PID, MSG, MSG_ENV)
```

Adding the `-fpermissive` compilation option turns this into a warning:

```bash
.../erlang/26.1.1/erts-14.1/include/erl_nif_api_funcs.h:652:47: warning: invalid conversion from ‘int’ to ‘ErlNifSelectFlags’ [-fpermissive]
  652 |     enif_select_x(ENV, E, ERL_NIF_SELECT_READ | ERL_NIF_SELECT_CUSTOM_MSG, \
      |                           ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                               |
      |                                               int
```

--- a/erts/emulator/beam/erl_nif_api_funcs.h
+++ b/erts/emulator/beam/erl_nif_api_funcs.h
@@ -640,13 +640,13 @@ static ERL_NIF_INLINE ERL_NIF_TERM enif_make_list9(ErlNifEnv* env,
 #  define enif_make_pid(ENV, PID) ((void)(ENV),(const ERL_NIF_TERM)((PID)->pid))
 #  define enif_compare_pids(A, B) (enif_compare((A)->pid,(B)->pid))
 #  define enif_select_read(ENV, E, OBJ, PID, MSG, MSG_ENV) \
-    enif_select_x(ENV, E, ERL_NIF_SELECT_READ | ERL_NIF_SELECT_CUSTOM_MSG, \
+    enif_select_x(ENV, E, (enum ErlNifSelectFlags)(ERL_NIF_SELECT_READ | ERL_NIF_SELECT_CUSTOM_MSG), \
                   OBJ, PID, MSG, MSG_ENV)
 #  define enif_select_write(ENV, E, OBJ, PID, MSG, MSG_ENV) \
-    enif_select_x(ENV, E, ERL_NIF_SELECT_WRITE | ERL_NIF_SELECT_CUSTOM_MSG, \
+    enif_select_x(ENV, E, (enum ErlNifSelectFlags)(ERL_NIF_SELECT_WRITE | ERL_NIF_SELECT_CUSTOM_MSG), \
                   OBJ, PID, MSG, MSG_ENV)
 #  define enif_select_error(ENV, E, OBJ, PID, MSG, MSG_ENV) \
-    enif_select_x(ENV, E, ERL_NIF_SELECT_ERROR | ERL_NIF_SELECT_CUSTOM_MSG, \
+    enif_select_x(ENV, E, (enum ErlNifSelectFlags)(ERL_NIF_SELECT_ERROR | ERL_NIF_SELECT_CUSTOM_MSG), \
                   OBJ, PID, MSG, MSG_ENV)
 
 #if SIZEOF_LONG == 8

